packages/x-data-grid-generator/package.json

```diff
@@ -37,7 +37,7 @@
     "@mui/x-data-grid-premium": "workspace:*",
     "chance": "^1.1.12",
     "clsx": "^2.1.1",
-    "lru-cache": "^10.4.3"
+    "lru-cache": "^11.0.0"
```

Comment:
@mui/xgrid can you confirm if my assumption that we can't bump this dependency is correct?
`@mui/x-data-grid-generator` has `engines.node` set to `>=14` and this major dependency bump now requires >=20 🤔 
Should we update `renovate` rule to include bumps only until v11?

Reply:
The CI is now green as MUI Core has bumped this dep to v11, but I'm still unsure about `@mui/data-grid-generator` working on node <20 with these changes.
@mui/xgrid Could you please check it? 🤔 🙏

Reply:
The generator package is not meant to be used by our users, so bumping the "engines.node" for it is not a breaking change.
On the other hand, I don't see a reason to upgrade this package. I prefer to keep using v10.

Reply:
Let's close this PR then. 👌 
If a new version is released, we can look into blocking updates with a `renovate` config. 👍

Reply:
This is now blocking typescript bump: https://github.com/mui/mui-x/pull/14571

Reply:
Should we reopen this PR and try bumping the package given this comment?
> The generator package is not meant to be used by our users, so bumping the "engines.node" for it is not a breaking change.

cc @cherniavskii